virttest/virt_vm.py

```diff
@@ -1110,62 +1051,66 @@ def print_guest_network_info():
             """
             Print guest network information into debug log file
             """
-            session = self.serial_login(internal_timeout, username, password, False)
+            session = None
             try:
+                session = self.serial_login(internal_timeout, username,
+                                            password)
                 out = session.cmd_output("ipconfig || ifconfig", timeout=60)
                 txt = ["Guest network status:\n %s" % out]
                 out = session.cmd_output("ip route || route print", timeout=60)
                 txt += ["Guest route table:\n %s" % out]
-                logging.debug("\n".join(txt))
+                logging.error("\n".join(txt))
+            except Exception, e:
+                logging.error("Can't get guest network status "
+                              "information, reason: %s", e)
             finally:
-                session.close()
+                if session:
+                    session.close()
 
+        error = None
         logging.debug("Attempting to log into '%s' (timeout %ds)",
                       self.name, timeout)
-        start = time.time()
+        start_time = time.time()
         try:
             self.wait_for_get_address(nic_index,
                                       timeout=timeout,
                                       ip_version=self.ip_version)
-        except Exception:
+        except Exception, e:
             self.verify_alive()
             print_guest_network_info()
             if not (serial or restart_network):
                 raise
-            if restart_network:
-                serial_login_args = (timeout, internal_timeout,
-                                     restart_network, username,
-                                     password, False)
-                session = self.wait_for_serial_login(*serial_login_args)
-                if serial:
-                    return session
-                session.close()
+            session = self.wait_for_serial_login(timeout, internal_timeout,
+                                                 restart_network, username,
+                                                 password, False)
+            if serial:
+                return session
+            session.close()
+            error = e
 
-        end = time.time()
-        # try to login if VM bootup really
-        timeout -= (end - start)
-        login_timeout = time.time() + timeout
-        while True:
+        # try to login if VM bootup really, at least once
+        not_tried = True
+        end_time = start_time + timeout
+        while time.time() < end_time or not_tried:
```

Comment:
could we use `utils_misc.wait_for()` here to reduce some code.

Reply:
I'm afraid not, because have to declare a wrapper function for that.